prov/gni/src/gnix_vc.c

```diff
@@ -281,34 +281,34 @@ static void __gnix_vc_pack_conn_req(char *sbuf,
 	      sizeof(gni_mem_handle_t);
 	assert(len <= GNIX_CM_NIC_MAX_MSG_SIZE);
 
-	memcpy(cptr, &rtype, sizeof(rtype));
+	*((uint8_t *)cptr) = rtype;
 	cptr += sizeof(rtype);
```

Comment:
In networking, the only thing you can do to avoid network noise is to flush the entire system and even that doesn't always do it. In my opinion, and you might ask others here as well, the only way to reliably measure performance gains in networking code is to take samples and look at the min/max/avg and variation of your samples. A reasonable number of samples will prevent outliers from affecting your statistics too much, especially if you disregard any sample outside of 2 or 3 standard deviations. The min, max, and average give pretty simple answers as to how performance changed, but keep in mind that a difference in min or max doesn't imply performance gain when the change in performance is small. The reason is that it could easily be attributed to the change as much as it could be attributed to a good run with little network noise. Average and variation are better indicators of performance gain since they are depend on the population, not a single sample.